app/connectors/EnrolmentStoreProxyConnector.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+import config.FrontendAppConfig
+import javax.inject.{Inject, Singleton}
+
+import com.google.inject.ImplementedBy
+import play.api.Logger
+import play.api.http.Status._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpGet}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class EnrolmentStoreProxyConnectorImpl @Inject()(val appConfig: FrontendAppConfig, val http: HttpClient)
+    extends EnrolmentStoreProxyConnector {
+
+  def checkExistingUTR(utr: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] =
+    http.GET(appConfig.checkUtrUrl(utr)).map { response =>
+      response.status match {
+        case OK         => true
+        case NO_CONTENT => false
+        case status =>
+          Logger.error(s"Enrolment Store Proxy returned status code: $status, body: ${response.body}")
```

Comment:
I think if you add a dependency on LoggerLike this can be injected

Reply:
discussed, not to be done in this ticket